mod_ci/controllers.py

```diff
@@ -763,7 +792,56 @@ def progress_reporter(test_id, token):
     return "FAIL"
 
 
+@mod_ci.route('/show_maintenance')
+@login_required
+@check_access_rights([Role.admin])
+@template_renderer('ci/maintenance.html', 404)
+def show_maintenance():
+    modes = MaintenanceMode.query.all()
+    return {
+        'modes': modes
+    }
+
+
+@mod_ci.route('/toggle_maintenance/<platform>/<status>')
+@login_required
+@check_access_rights([Role.admin])
+def toggle_maintenance(platform, status):
+    db_mode = MaintenanceMode.query.filter(MaintenanceMode.platform ==
+                                           platform).first()
+    if db_mode is None:
+        status = 'failed'
+        message = 'Platform Not found'
+    elif status == 'True':
+        db_mode.mode = 'True'
+        g.db.commit()
+        status = 'success'
+        message = platform + ' platform is in maintenance mode'
+    elif status == 'False':
+        db_mode.mode = 'False'
+        g.db.commit()
+        status = 'success'
+        message = platform + ' platform is in active mode'
+    else:
+        status = 'failed'
+        message = 'No Change'
+    return jsonify({
+        'status': status,
+        'message': message
+    })
+
+
 @mod_ci.route('/maintenance-mode/<platform>')
 def in_maintenance_mode(platform):
-    # TODO: fetch status from DB if #85 is implemented
-    return "False"
+    platforms = TestPlatform.list_all()
+    if platform not in platforms:
+        return 'ERROR'
+    db_mode = MaintenanceMode.query.filter(MaintenanceMode.platform ==
```

Comment:
platform is a string here, you most likely can't convert that to a platform instance. Is this tested?